Source/Renderer/ComputeEngine.js

```diff
@@ -0,0 +1,194 @@
+        PrimitiveType,
+        ViewportQuadVS,
+        BufferUsage,
+        ClearCommand,
+        DrawCommand,
+        Framebuffer,
+        RenderState,
+        ShaderProgram,
+        VertexArray) {
+    "use strict";
+
+    /**
+     * @private
+     */
+    var ComputeEngine = function(context) {
+        this._context = context;
+        this._viewportQuadVertexArray = undefined;
+    };
+
+    function createViewportQuadVertexArray(computeEngine) {
+        var vertexArray = computeEngine._viewportQuadVertexArray;
+
+        if (!defined(vertexArray)) {
+            var geometry = new Geometry({
+                attributes : {
+                    position : new GeometryAttribute({
+                        componentDatatype : ComponentDatatype.FLOAT,
+                        componentsPerAttribute : 2,
+                        values : [
+                            -1.0, -1.0,
+                            1.0, -1.0,
+                            1.0,  1.0,
+                            -1.0,  1.0
+                        ]
+                    }),
+
+                    textureCoordinates : new GeometryAttribute({
+                        componentDatatype : ComponentDatatype.FLOAT,
+                        componentsPerAttribute : 2,
+                        values : [
+                            0.0, 0.0,
+                            1.0, 0.0,
+                            1.0, 1.0,
+                            0.0, 1.0
+                        ]
+                    })
+                },
+                // Workaround Internet Explorer 11.0.8 lack of TRIANGLE_FAN
+                indices : new Uint16Array([0, 1, 2, 0, 2, 3]),
+                primitiveType : PrimitiveType.TRIANGLES
+            });
+
+            vertexArray = VertexArray.fromGeometry({
+                context : computeEngine._context,
+                geometry : geometry,
+                attributeLocations : {
+                    position : 0,
+                    textureCoordinates : 1
+                },
+                bufferUsage : BufferUsage.STATIC_DRAW,
+                interleave : true
+            });
+
+            computeEngine._viewportQuadVertexArray = vertexArray;
+        }
+
+        return vertexArray;
+    }
+
+    var viewportQuadAttributeLocations = {
+        position : 0,
+        textureCoordinates : 1
+    };
+
+    function createFramebuffer(context, texture) {
+        var fbo = new Framebuffer({
+            context : context,
+            colorTextures : [texture]
+        });
+        fbo.destroyAttachments = false;
+        return fbo;
+    }
+
+    function createViewportQuadShader(context, fragmentShaderSource) {
+        return ShaderProgram.fromCache({
+            context : context,
+            vertexShaderSource : ViewportQuadVS,
+            fragmentShaderSource : fragmentShaderSource,
+            attributeLocations : viewportQuadAttributeLocations
+        });
+    }
+
+    function createRenderState(width, height)
```

Comment:
Actually, we only call this function in one place; I would just put the code inline.